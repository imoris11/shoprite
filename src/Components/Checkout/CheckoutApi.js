export const getTaxApi = async (url) => {
  let response = await fetch(url);
  return await response.json();
}

export const makeOrderApi = async (url, data, token) => {
  let response = await fetch(url, {
    method:'POST',
    body: JSON.stringify(data),
    headers:{
      'Accept':'application/json',
      'content-type':'application/json',
      'user-key':token
    }
  });
  return await response.json();
}

export const chargeOrderApi = async (url, data) => {
  let response = await fetch(url, {
    method:'POST',
    headers:{
      'content-type':'application/json',
      'Accept':'application/json',
    },
    body:JSON.stringify(data)
  });
  return;
}
