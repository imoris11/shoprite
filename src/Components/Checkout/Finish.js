import React, {Component} from 'react';
import { Row, Col } from 'react-bootstrap';
import Button from '@material-ui/core/Button';
import {Link} from 'react-router-dom';
export default class Delivery extends Component {
  render () {
    return (
      <div>
        <Row>
          <Col>
            <img alt="rocket" src={require('../../Resources/images/icons-rocket.png')}/>
          </Col>
          <Col>
            <h3 style={{marginTop:30}}>Success!</h3>
            <p>Your items will be shipped shortly. Check your email for more details.</p>
          </Col>
        </Row>
        <Row >
          <Col >
            <Row className='float-right'>
              <Col>
                <Link to="/">
                  <Button  variant='contained' color="secondary">
                    Go Home
                  </Button>
                </Link>
              </Col>
            </Row>
          </Col>
        </Row>
      </div>

    )
  }
}
