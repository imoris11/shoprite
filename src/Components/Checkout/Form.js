import React, {Component} from 'react';
import StripeCheckout from 'react-stripe-checkout';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total:0
    }
    this.submit = this.submit.bind(this);
  }
  componentDidMount () {
    let total = Number(localStorage.getItem('total'));
    this.setState({total})
  }

  async submit(token) {
    this.props.isLoading(true);
    this.fetchTaxInfo(token);
  }
  fetchTaxInfo (stripeToken) {
    //Get tax info, and calculate initial costs
    fetch('https://backendapi.turing.com/tax')
    .then(response => response.json())
    .then(result => {
      if (result) {
        this.setState({tax_id: result[0].tax_id});
        this.makeOrder(stripeToken, result[0].tax_id);
      }
    })
    .catch(error => {
      console.log("Oops!, encountered an error");
    })
  }
  makeOrder = (stripeToken, tax_id) => {
    //Get required details for API
    let cart_id = localStorage.getItem('cart_id');
    let customer_id = localStorage.getItem('customer_id');
    let shipping_id = localStorage.getItem('shipping_id');
    let token = localStorage.getItem('token');
    let data = {
      customer_id:customer_id,
      shipping_id:shipping_id,
      cart_id:cart_id,
      tax_id:tax_id
    }
    //Make API call to orders
    fetch('https://backendapi.turing.com/orders', {
      method:'POST',
      body: JSON.stringify(data),
      headers:{
        'Accept':'application/json',
        'content-type':'application/json',
        'user-key':token
      }
    })
    .then(response => response.json())
    .then(result => {
      //Make charge on the card provided
      this.chargeOrder(stripeToken, result);
    })
    .catch(error => {
      console.log(error);
    })
  }
  chargeOrder (stripeToken, orderId) {
    //Get total amount to be charged to card
    let total = Number(localStorage.getItem('total'));
    let data = {
      stripeToken: stripeToken.id,
      order_id: Number(orderId.order_id),
      description: "Payment for items",
      amount: Math.floor(total * 100),
    }
    //Make API call to stripe/charge with token id retrieved from StripeCheckout
    fetch('https://backendapi.turing.com/stripe/charge', {
      method:'POST',
      headers:{
        'content-type':'application/json',
        'Accept':'application/json',
      },
      body:JSON.stringify(data)
    })
    .then(response => {
      return response.json();
    })
    .then(res => {
      //Proceed to next on success
      this.props.onNext();
    })
    .catch(error => {
      console.log(error);
    })

  }
  render() {
    return (
      <div className="checkout">
        <StripeCheckout
          token={this.submit}
          name="SHOPMATE" // the pop-in header title
          description="Amazing clothings, amazing prices"
          amount={this.state.total * 100} // cents
          currency="USD"
          stripeKey="pk_test_NcwpaplBCuTL6I0THD44heRe"
        />
      </div>
    );
  }
}

export default Form;
