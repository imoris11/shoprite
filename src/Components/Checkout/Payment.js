import React, {Component} from 'react';
import StripeCheckout from './Form';
import { Row, Col } from 'react-bootstrap';
import '../../Resources/css/payment.css';
import { getTotalCostApi } from './CheckoutApi';
import LinearIndeterminate from '../Extra/Loading';
export default class Payment extends Component {
  state = {
    payment:'card',
    total:0,
  }
  componentDidMount () {
    this.getTotalCost();
  }
  async getTotalCost () {
    let { total } = this.state;
    //Get total cost from online cart
    let cart_id = localStorage.getItem('cart_id');
    let url = 'https://backendapi.turing.com/shoppingcart/totalAmount/'+cart_id;
    let res = await getTotalCostApi(url);
    total = Number(res.total_amount);

    this.setState({total});
  }
  onNext = () => this.props.onNext();
  setIsLoading = () => this.setState({loading:true})
  render () {
    let {payment} = this.state;
    return (
      <Row>
        <Col md={{span:8, offset:2}}>
          <Row>
            <Col onClick={()=>this.setState({payment:'card'})} className='text-center payment-method' style={{backgroundColor: payment === 'card' && 'lightblue'}} >
              <img src={require('../../Resources/images/logos-visa.png')} className='payment-image' alt="Visa" />
              <img src={require('../../Resources/images/logos-mastercard.png')} className='payment-image' alt="Mastercard" />
            </Col>
            <Col onClick={()=>this.setState({payment:'paypal'})} className='text-center payment-method' style={{backgroundColor: payment === 'paypal' && 'lightblue'}}>
              <img src={require('../../Resources/images/logos-PayPal.png')} className='payment-image' alt="Paypal" />
            </Col>
          </Row>
          {!this.state.loading && <Col>
            <h6 >Total: ${this.state.total.toFixed(2)}</h6>
            {payment === 'card' &&
            <StripeCheckout isLoading={this.setIsLoading} onNext={this.onNext} />}
          </Col>}
          {this.state.loading && <LinearIndeterminate />}
        </Col>
      </Row>
    )
  }
}
