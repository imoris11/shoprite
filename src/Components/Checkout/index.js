import React, { Component } from 'react';
import { Col } from 'react-bootstrap';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Registration from '../Registration';
import Contact from '../Profile/Contact';
import Payment from './Payment';
import Finish from './Finish';
import Cart from '../Cart';

export default class Checkout extends Component {
  state = {
    steps: ['Confirmation', 'Contact Information', 'Payment', 'Finish'],
    activeStep:0,
    showDialog:false
  }
  componentDidMount () {
    this.confirmUserStatus ()
  }
  confirmUserStatus = () => {
    let token = localStorage.getItem('token');
    if (token) {
      this.setState({token, loggedIn:true});
    }else{
      this.setState({showDialog:true})
    }
  }
  getStepperContent (activeStep) {
    //Determine what contnt to show basd on activeStep
    switch (activeStep) {
      case 0:
        return <Cart onNext={this.onNext} checkout={true} />
        break;
      case 1:
        return <Contact onNext={this.onNext} />
        break;
      case 2:
        return <Payment onNext={this.onNext} />
        break;
      case 3:
        return <Finish />
        break;
      default:
        return null;
    }
  }
  onNext = () => {
    this.setState(prevState => ({activeStep:prevState.activeStep + 1}));
  }
  onBack = () => {
    this.setState(prevState => ({activeStep:prevState.activeStep - 1}));
  }
  handleClose = () => {
    //Do nothing. Don't close unless logged in
  }
  render () {
    let { steps, activeStep } = this.state;
    return (
      <Col style={{marginTop:30}}>
        <Col md={{span:10, offset:1}}>
          <Stepper activeStep={activeStep}>
          {steps.map((label, index) => {
            const stepProps = {};
            const labelProps = {};
            return (
              <Step key={label} {...stepProps}>
                <StepLabel {...labelProps}>{label}</StepLabel>
              </Step>
            );
          })}
        </Stepper>
        {this.getStepperContent(activeStep)}
        </Col>
        <Registration showDialog={this.state.showDialog} handleClose={this.handleClose} />
      </Col>

    )
  }
}
