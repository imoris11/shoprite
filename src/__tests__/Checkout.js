import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import Checkout from '../Components/Checkout';

describe('Test checkout component', () => {

   it('renders without crashing', () => {
      shallow(<Checkout />);
    });

    it('renders correctly', ()=> {
      const component = shallow(<Checkout />);
      expect(component).toMatchSnapshot();
    });

});
