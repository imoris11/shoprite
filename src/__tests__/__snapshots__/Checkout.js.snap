// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test checkout component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": Array [
        <ForwardRef
          as="div"
          md={
            Object {
              "offset": 1,
              "span": 10,
            }
          }
        >
          <WithStyles(Stepper)
            activeStep={0}
          >
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                Confirmation
              </WithStyles(StepLabel)>
            </WithStyles(Step)>
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                Contact Information
              </WithStyles(StepLabel)>
            </WithStyles(Step)>
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                Payment
              </WithStyles(StepLabel)>
            </WithStyles(Step)>
            <WithStyles(Step)>
              <WithStyles(StepLabel)>
                Finish
              </WithStyles(StepLabel)>
            </WithStyles(Step)>
          </WithStyles(Stepper)>
          <WithStyles(Cart)
            checkout={true}
            onNext={[Function]}
          />
        </ForwardRef>,
        <CustomizedDialog
          handleClose={[Function]}
          showDialog={true}
        />,
      ],
      "style": Object {
        "marginTop": 30,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": Array [
            <WithStyles(Stepper)
              activeStep={0}
            >
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Confirmation
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Contact Information
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Payment
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Finish
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
            </WithStyles(Stepper)>,
            <WithStyles(Cart)
              checkout={true}
              onNext={[Function]}
            />,
          ],
          "md": Object {
            "offset": 1,
            "span": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeStep": 0,
              "children": Array [
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Confirmation
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>,
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Contact Information
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>,
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Payment
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>,
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Finish
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Confirmation",
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(StepLabel)>
                    Confirmation
                  </WithStyles(StepLabel)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Confirmation",
                  },
                  "ref": null,
                  "rendered": "Confirmation",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Contact Information",
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(StepLabel)>
                    Contact Information
                  </WithStyles(StepLabel)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Contact Information",
                  },
                  "ref": null,
                  "rendered": "Contact Information",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Payment",
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(StepLabel)>
                    Payment
                  </WithStyles(StepLabel)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Payment",
                  },
                  "ref": null,
                  "rendered": "Payment",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Finish",
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(StepLabel)>
                    Finish
                  </WithStyles(StepLabel)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Finish",
                  },
                  "ref": null,
                  "rendered": "Finish",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checkout": true,
              "onNext": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleClose": [Function],
          "showDialog": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
      },
      "displayName": "Col",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": Array [
          <ForwardRef
            as="div"
            md={
              Object {
                "offset": 1,
                "span": 10,
              }
            }
          >
            <WithStyles(Stepper)
              activeStep={0}
            >
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Confirmation
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Contact Information
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Payment
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
              <WithStyles(Step)>
                <WithStyles(StepLabel)>
                  Finish
                </WithStyles(StepLabel)>
              </WithStyles(Step)>
            </WithStyles(Stepper)>
            <WithStyles(Cart)
              checkout={true}
              onNext={[Function]}
            />
          </ForwardRef>,
          <CustomizedDialog
            handleClose={[Function]}
            showDialog={true}
          />,
        ],
        "style": Object {
          "marginTop": 30,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <WithStyles(Stepper)
                activeStep={0}
              >
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Confirmation
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Contact Information
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Payment
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>
                <WithStyles(Step)>
                  <WithStyles(StepLabel)>
                    Finish
                  </WithStyles(StepLabel)>
                </WithStyles(Step)>
              </WithStyles(Stepper)>,
              <WithStyles(Cart)
                checkout={true}
                onNext={[Function]}
              />,
            ],
            "md": Object {
              "offset": 1,
              "span": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeStep": 0,
                "children": Array [
                  <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                      Confirmation
                    </WithStyles(StepLabel)>
                  </WithStyles(Step)>,
                  <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                      Contact Information
                    </WithStyles(StepLabel)>
                  </WithStyles(Step)>,
                  <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                      Payment
                    </WithStyles(StepLabel)>
                  </WithStyles(Step)>,
                  <WithStyles(Step)>
                    <WithStyles(StepLabel)>
                      Finish
                    </WithStyles(StepLabel)>
                  </WithStyles(Step)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Confirmation",
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(StepLabel)>
                      Confirmation
                    </WithStyles(StepLabel)>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Confirmation",
                    },
                    "ref": null,
                    "rendered": "Confirmation",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Contact Information",
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(StepLabel)>
                      Contact Information
                    </WithStyles(StepLabel)>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Contact Information",
                    },
                    "ref": null,
                    "rendered": "Contact Information",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Payment",
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(StepLabel)>
                      Payment
                    </WithStyles(StepLabel)>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Payment",
                    },
                    "ref": null,
                    "rendered": "Payment",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Finish",
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(StepLabel)>
                      Finish
                    </WithStyles(StepLabel)>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Finish",
                    },
                    "ref": null,
                    "rendered": "Finish",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checkout": true,
                "onNext": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleClose": [Function],
            "showDialog": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
        },
        "displayName": "Col",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
